tests/financial/test_national_stock_exchange_of_india.py

```diff
@@ -0,0 +1,742 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.financial.national_stock_exchange_of_india import (
+    NSE,
+    XNSE,
+    NationalStockExchangeOfIndia,
+)
+from tests.common import CommonFinancialTests
+
+
+class TestNationalStockExchangeOfIndia(CommonFinancialTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(2001, 2050)
+        super().setUpClass(NationalStockExchangeOfIndia, years=years)
+        cls.nonobs_no_estimated_holidays = NationalStockExchangeOfIndia(
+            observed=False, years=years, islamic_show_estimated=False
+        )
+
+    def test_market_aliases(self):
+        self.assertAliases(NationalStockExchangeOfIndia, NSE, XNSE)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(NationalStockExchangeOfIndia(years=2000))
+
+    def test_republic_day(self):
+        name = "Republic Day"
+        self.assertNonObservedHolidayName(name, (f"{year}-01-26" for year in range(2001, 2050)))
+        no_dt = ("2025-01-26",)
+        self.assertNoHolidayName(name, no_dt)
```

Comment:
```suggestion
        self.assertNonObservedHolidayName(
            "Republic Day", (f"{year}-01-26" for year in range(2001, 2050))
        )
        self.assertNoHoliday(
            "2025-01-26",
        )
```
And same approach for rest of fixed-date Gregorian holidays.

Reply:
updated.